extension/docs/time_weighted_average.md

```diff
@@ -0,0 +1,254 @@
+# Time Weighted Average [<sup><mark>experimental</mark></sup>](/extension/docs/README.md#tag-notes)
+
+> [Description](#time-weighted-average-description)<br>
+> [API](#time-weighted-average-api) <br>
+> [Sample Usage](time-weighted-average-examples)<br>
+> [Interpolation Methods Details](#time-weight-methods)<br>
+> [Notes on Parallelism and Ordering](#time-weight-ordering)<br>
+
+## Description [](time-weighted-average-description)
+
+Time weighted averages are commonly used in cases where a time series is not evenly sampled, so a traditional average will give misleading results. Consider a voltage sensor that sends readings once every 5 minutes or whenever the value changes by more than 1 V from the previous reading. If the results are generally stable, but with some quick moving transients, a simple average over all of the points will tend to over-weight the transients instead of the stable readings. A time weighted average weights each value by the duration over which it occured based on the points around it and produces correct results for unevenly spaced series. 
+
+Timescale Analytics' time weighted average is implemented as an aggregate which weights each value either using a last observation carried forward (LOCF) approach or a linear interpolation approach ([see interpolation methods](#time-weight-methods)). While the aggregate is not parallelizable, it is supported with [continuous aggregation](https://docs.timescale.com/latest/using-timescaledb/continuous-aggregates). 
+
+Additionally, see the notes on 
```

Comment:
Assuming there's more coming here?

Reply:
yep, I noticed that right after I pushed :)